prometheus_hardware_exporter/collector.py

```diff
@@ -903,8 +1008,111 @@ def fetch(self) -> List[Payload]:
                     )
                 )
 
-        return payloads
+        for system_id, curr_processor_data in metrics["processor_data"].items():
```

Comment:
this is still too long. each for loop can have its own function with a meaningful name

Reply:
Hmm, alright. Let me create individual functions for adding each of the individual metrics to the payload.

Reply:
I'm okay with this function. Although it's long, but the structure is simple, we add the payload one by one. It's not difficult to understand what we want to do in this function. But from unit test's point of view, it's may be easier to write unit test when we split into small function. I will let Ashley to make the final decision.